examples/abi/res/abi.json

```
@@ -15,57 +15,116 @@
         "params": [
           {
             "name": "a",
+            "serialization_type": "json",
             "type_schema": {
               "$ref": "#/definitions/Pair"
-            },
-            "serialization_type": "json"
+            }
           },
           {
             "name": "b",
+            "serialization_type": "json",
             "type_schema": {
               "$ref": "#/definitions/Pair"
-            },
-            "serialization_type": "json"
+            }
           }
         ],
         "result": {
+          "serialization_type": "json",
           "type_schema": {
             "$ref": "#/definitions/Pair"
+          }
+        }
+      },
+      {
+        "name": "add_borsh",
+        "is_view": true,
+        "params": [
+          {
+            "name": "a",
+            "serialization_type": "borsh",
+            "type_schema": {
+              "declaration": "Pair",
+              "definitions": {
+                "Pair": {
+                  "Struct": {
+                    "fields": {
+                      "UnnamedFields": [
+                        "u32",
+                        "u32"
+                      ]
+                    }
+                  }
+                }
+              }
+            }
           },
-          "serialization_type": "json"
+          {
+            "name": "b",
+            "serialization_type": "borsh",
+            "type_schema": {
+              "declaration": "Pair",
+              "definitions": {
+                "Pair": {
+                  "Struct": {
+                    "fields": {
+                      "UnnamedFields": [
+                        "u32",
+                        "u32"
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          }
+        ],
+        "result": {
+          "serialization_type": "borsh",
+          "type_schema": {
+            "declaration": "Pair",
+            "definitions": {
+              "Pair": {
+                "Struct": {
+                  "fields": {
+                    "UnnamedFields": [
+                      "u32",
+                      "u32"
+                    ]
+                  }
+                }
```

Intuitively, feels like this should be
```suggestion
                "Struct": [
                      "u32",
                      "u32"
                    ]
```

And I can't see why you wouldn't be able to do this because the struct is either a JSON map, an array, or empty (empty map or null, idk what it's used for), am I missing something?

---

Hmm yeah that's completely fair. I will see if I can use remote derivation with custom serde attributes to control the format a bit more.

---

Done! This introduced even more complexity to the code, but at least it seems to work. This definitely needs some tests though, will add them next week.